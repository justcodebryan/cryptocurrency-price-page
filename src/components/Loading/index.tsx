type LoadingProps = {
  loading?: boolean
}

const Loading = ({ loading = false }: LoadingProps) => {
  return <>{loading ? 'Loading...' : ''}</>
}

export default Loading
