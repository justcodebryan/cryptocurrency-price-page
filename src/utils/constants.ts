import { NoopFn } from '@/types'

export const NOOP: NoopFn = () => void 0

export const DEFAULT_MODAL_CONTAINER_ID = 'modal-container'

export const DEFAULT_POLLING_INTERVAL = 10 * 1000
