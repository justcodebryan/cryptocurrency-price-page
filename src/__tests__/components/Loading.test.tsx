import Loading from '@/components/Loading'
import { render, screen } from '@testing-library/react'
import { toJSON } from '@/utils/test-utils'
import renderer from 'react-test-renderer'

describe('Loading Test Suit', () => {
  test('Loading Snapshot', () => {
    const component = renderer.create(<Loading />)
    const tree = toJSON(component)
    expect(tree).toMatchSnapshot()
  })

  test('Should show the content at the start', () => {
    render(<Loading loading={true} />)
    expect(screen.getByText(/Loading.../i)).toBeDefined()
  })
})
