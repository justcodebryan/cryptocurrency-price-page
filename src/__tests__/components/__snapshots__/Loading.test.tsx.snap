// Vitest Snapshot v1

exports[`Loading Test Suit > Loading Snapshot 1`] = `null`;
